packages/p2p-media-loader-shaka/src/services/loading-handler.ts

```diff
@@ -48,79 +54,63 @@ export class LoadingHandler implements LoadingHandlerInterface {
     }
 
     const loading = this.defaultLoad();
-    if (
-      requestType === RequestType.MANIFEST &&
-      this.streamInfo.protocol === "hls"
-    ) {
-      void this.handleStreamLoading(url, loading.promise);
+    if (requestType === RequestType.MANIFEST) {
+      void this.handleManifestLoading(url, loading.promise);
     }
     return loading;
   }
 
-  private async handleStreamLoading(
+  private async handleManifestLoading(
     streamUrl: string,
-    loadingPromise: Promise<unknown>
+    loadingPromise: Promise<Response>
   ) {
-    if (!this.segmentManager.urlStreamMap.has(streamUrl)) return;
-    await loadingPromise;
-    // Waiting for the playlist to be parsed
-    setTimeout(() => this.segmentManager.updateHlsStreamByUrl(streamUrl), 0);
+    if (
+      this.streamInfo.protocol === "hls" &&
+      !!this.core.getStreamByUrl(streamUrl)
+    ) {
+      // loading HLS playlist manifest
+      await loadingPromise;
+      // Waiting for the playlist to be parsed
+      setTimeout(() => this.segmentManager.updateHlsStreamByUrl(streamUrl), 0);
```

Comment:
`updateHlsStreamByUrl` calls `this.core.getStreamByUrl(streamUrl)` again. We can probably optimize.